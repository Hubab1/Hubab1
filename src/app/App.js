import { connect } from 'react-redux';
import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';
import queryString from 'query-string';

import Main from 'app/Main';
import BadRoute from 'components/common/BadRoute';
import { basenameReceived } from 'reducers/site-config';

export class App extends Component {
    componentDidMount () {
        const basename = window.location.pathname.split('/')[1];
        const siteConfig = {};
        siteConfig.basename = basename;

        const params = queryString.parse(window.location.search);
        siteConfig.hash = params.v;
        
        this.props.basenameReceived(siteConfig);
    }

    render() {
        if (!this.props.basename || this.props.basename === '/') return <BadRoute/>;
        return (
            <div className="App">
                <BrowserRouter basename={this.props.basename}>
                    <Main/>
                </BrowserRouter>
            </div>
        );
    }
}

const mapStateToProps = state => ({basename: state.siteConfig.basename});
const mapDispatchToProps = {basenameReceived};
export default connect(mapStateToProps,mapDispatchToProps)(App);
