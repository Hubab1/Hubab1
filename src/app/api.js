import auth from 'utils/auth';
import fetch from 'app/fetch';
export const MOCKY = !!process.env.REACT_APP_MOCKY;

export const CHUCK_BASE_URL = process.env.REACT_APP_CHUCK_DOMAIN;

export function chuck(path) {
    return `${CHUCK_BASE_URL}/api/onlineleasing${path}`;
}

const getWithHeaders = (url) => fetch(url, {
    headers: {
        "Content-Type": "application/json",
        "Accept": "application/json"
    }
}).then(res => res.json());

const CHUCK_BASE_LEASE_SETTINGS = (communityId) => `${CHUCK_BASE_URL}/api/onlineleasing/communities/${communityId}/`;
const CHUCK_PERSONALIZED_LEASE_SETTINGS = (communityId, hash) => `${CHUCK_BASE_URL}/api/onlineleasing/communities/${communityId}/personalized/${hash}`;

const API = {};

API.fetchConfiguration = (communityId) => {
    return getWithHeaders(CHUCK_BASE_LEASE_SETTINGS(communityId));
};

API.fetchPrivacyPolicy = (communityId) => {
    return fetch(chuck('/privacy-policy'));
};

API.fetchPersonalizedInfo = (communityId, hash) => {
    return getWithHeaders(CHUCK_PERSONALIZED_LEASE_SETTINGS(communityId, hash));
};

API.postPageComplete = (page) => {
    return fetch(chuck(`/page-complete/`), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify({ page })
    }).then(res => res.json());
};

API.fetchAvailableUnits = () => {
    if (MOCKY) return Promise.resolve([
        {id: 1, unit_number: '1B', price: 1540.50},
        {id: 2, unit_number: '1C', price: 1999.99},
        {id: 3, unit_number: '1D', price: 5000.00},
        {id: 4, unit_number: '1E', price: 2450.99},
        {id: 5, unit_number: '2E', price: 2450.99},
        {id: 6, unit_number: '2A', price: 1540.50},
        {id: 7, unit_number: '2B', price: 1999.99},
        {id: 8, unit_number: '2C', price: 5000.00},
        {id: 9, unit_number: '2D', price: 2450.99},
        {id: 10, unit_number: '2E', price: 2450.99},
    ])
    return fetch(chuck('/available-units/'), {
        method: 'GET',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        }
    }).then(res => res.json());
};

API.patchApplication = (data) => {
    return fetch(chuck('/application/'), {
        method: 'PATCH',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
}

API.putApplicant = (data) => {
    return fetch(chuck('/applicant/'), {
        method: 'PUT',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
}

API.fetchRenterProfile = () => {
    return fetch(chuck('/application/'), {
        method: 'GET',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        }
    }).then(res => res.json());
};

API.fetchApplicant = () => {
    return fetch(chuck('/applicant/'), {
        method: 'GET',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        }
    }).then(res => res.json());
};

API.login = (email, password, communityId) => {
    return fetch(chuck(`/communities/${communityId}/login/`), {
        method: 'POST',
        body: JSON.stringify({ email, password })
    }).then(res => res.json());
};

API.register = (data, leaseSettingsId, hash) => {
    let registerUrl = `/communities/${leaseSettingsId}/register/`;
    if (hash) { registerUrl = registerUrl.concat(`?v=${hash}`); }
    return fetch(chuck(registerUrl), {
        method: 'POST',
        body: JSON.stringify(data)
    }).then(res => res.json());
};

API.acceptTerms = (data) => {
    return fetch(chuck('/terms-accepted/'), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
};

API.passwordResetRequest = (phone_number, lease_settings_id) => {
    return fetch(chuck('/password-reset/'), {
        method: 'POST',
        body: JSON.stringify({ phone_number, lease_settings_id })
    }).then(res => res.json());
};

API.passwordResetVerification = (phone_number, code, lease_settings_id) => {
    return fetch(chuck('/password-reset-verification/'), {
        method: 'POST',
        body: JSON.stringify({ phone_number, code, lease_settings_id })
    }).then(res => res.json());
}

API.passwordReset = (password, token) => {
    return fetch(chuck('/password-change/'), {
        method: 'PUT',
        body: JSON.stringify({password}),
        headers: {
            "Content-Type": "application/json",
            'AUTHORIZATION': `token ${token}`,
        }
    }).then(res => res.json());
}

API.inviteGuarantor = (data) => {
    return fetch(chuck('/guarantors/'), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
}

API.createFinicityUrl = () => {
    return fetch(chuck('/generate-finicity-link/'), {
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    }).then(res => res.json());}

API.generateFinicityReports = () => {
    return fetch(chuck('/generate-finicity-reports/'), {
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    }).then(res => res.json());
}

API.embeddedSigningUrl = () => {
    return fetch(chuck('/embedded-signing-url/'), {
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    }).then(res => res.json());
}
API.leaseDocumentUrl = () => {
    return fetch(chuck('/lease-document-url/'), {
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    }).then(res => res.json());
}

API.fetchFinicityReports = () => {
    return fetch(chuck('/fetch-finicity-reports/'), {
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    });
}

API.stripePayment = (data) => {
    return fetch(chuck('/payment/'), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
}

API.createIncomeStream = (data) => {
    return fetch(chuck('/incomestreams/'), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
}

API.updateInvitee = (data, inviteeId) => {
    return fetch(chuck(`/invitees/${inviteeId}/`), {
        method: 'PUT',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json());
}

API.postPassthrough = data => (
    fetch(chuck('/passthrough/'), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => res.json())
)

API.fetchPaymentOptions = () => {
    return fetch(chuck('/payment-options/'), {
        method: 'GET',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    }).then(res => {
        return res.json()
    });
}

API.getCurrentFlatQuote = (data) => {
    return fetch(chuck('/payment-breakdown/'), {
        method: 'POST',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
        body: JSON.stringify(data)
    }).then(res => {
        return res.json()
    });
}

API.getDenialDecisionDetails = () => {
    return fetch(chuck('/applicant/denial-decision/'), {
        method: 'GET',
        headers: {
            Authorization: `Token ${auth.getToken()}`
        },
    }).then(res => {
        return res.json()
    });
}

export default API;
