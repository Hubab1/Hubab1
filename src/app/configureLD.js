import ldRedux from 'ld-redux';

import store from 'reducers/store';

const flags = {
    'agent-request-more-income-when-no-income': false,
};

ldRedux.init({
    clientSideId: process.env.REACT_APP_LD_CLIENT_KEY,
    dispatch: store.dispatch,
    flags,
});
