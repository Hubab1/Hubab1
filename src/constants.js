export const community = {
    name: '555 Waverly'
};

export const Routes = {
    WELCOME: 'welcome',
    PROFILE: 'profile',
    LOGIN: 'login',
    SIGNUP: 'signup',
    TOS: 'terms',
    PROFILE_OPTIONS: 'profile/options',
    PETS: 'profile/pets'
}

export const ALL_ROUTES = [
    Routes.LOGIN, Routes.TOS, Routes.PROFILE_OPTIONS, Routes.PETS
]
