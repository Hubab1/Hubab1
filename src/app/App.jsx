import { connect } from 'react-redux';
import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';
import queryString from 'query-string';
import PropTypes from 'prop-types';

import { DOES_NOT_EXIST } from 'constants/constants';
import { basenameReceived } from 'reducers/site-config';

import Main from './Main';
import ScrollToTop from 'common-components/ScrollToTop/ScrollToTop';
import NotFoundPage from 'pages/NotFound';

export class App extends Component {
    componentDidMount() {
        const basename = window.location.pathname.split('/')[1];
        const siteConfig = {};
        siteConfig.basename = basename;

        const params = queryString.parse(window.location.search);
        siteConfig.hash = params.v;

        console.log({
            'window.location.pathname: ': window.location.pathname,
            basename
        })

        this.props.basenameReceived(siteConfig);
    }

    badRoute() {
        return (
            !this.props.basename ||
            this.props.basename === '/' ||
            this.props.configuration.error_type === DOES_NOT_EXIST
        );
    }

    render() {
        if (this.badRoute()) return <NotFoundPage />;

        return (
            <div className="App">
                <BrowserRouter basename={this.props.basename}>
                    <ScrollToTop />
                    <Main />
                </BrowserRouter>
            </div>
        );
    }
}

App.propTypes = {
    basenameReceived: PropTypes.func.isRequired,
    basename: PropTypes.string.isRequired,
    configuration: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({ basename: state.siteConfig.basename, configuration: state.configuration });
const mapDispatchToProps = { basenameReceived };
export default connect(mapStateToProps, mapDispatchToProps)(App);
