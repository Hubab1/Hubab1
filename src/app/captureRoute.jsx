import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { currentRouteReceived } from 'reducers/site-config';

import { MOCKY } from 'app/api';
import { fetchApplicant } from 'reducers/applicant';
import { fetchRenterProfile } from 'reducers/renter-profile';
import { ROUTES } from 'app/constants';

// Second param is deprecated in favor of static param 'route'
export default function captureRoute(WrappedComponent, route) {
    // Make sure route is a top level page route! or else you will break relative routing.
    route = WrappedComponent.route || route;
    class Component extends React.Component {
        constructor (props) {
            super(props);
            props.currentRouteReceived(route);
        }
        render() {
            return (
                <WrappedComponent
                    _navigate={
                        async (route) => {
                            if (!MOCKY) {
                                this.props.fetchApplicant();
                                await this.props.fetchRenterProfile();
                            }
                            if (this.props.unitAvailable === false) {
                                return this.props.history.push(ROUTES.UNIT_UNAVAILABLE);
                            } else {
                                return this.props.history.push(route);
                            }
                        }
                    }
                    {...this.props}
                />);
        }
    }

    Component.propTypes = {
        currentRouteReceived: PropTypes.func,
        fetchApplicant: PropTypes.func,
        fetchRenterProfile: PropTypes.func,
        history: PropTypes.object,
    };

    return connect(
        null,
        {currentRouteReceived, fetchApplicant, fetchRenterProfile})
    (Component);
}
