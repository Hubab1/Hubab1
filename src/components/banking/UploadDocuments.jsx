import React from "react";
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import withRelativeRoutes from 'app/withRelativeRoutes';

import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import ActionButton from 'components/common/ActionButton/ActionButton';

import { ROUTES } from 'app/constants';
import { FINANCIAL_STREAM_INCOME, FINANCIAL_STREAM_ASSET } from 'app/constants';
import { P } from 'assets/styles';


export class UploadDocuments extends React.Component {
    state = {
        selectedDocumentIndex: null,
        selectedDocument: null,
    };

    componentDidUpdate (prevProps) {
        if (prevProps.incomeOrAssetType !== this.props.incomeOrAssetType){
            this.setState({
                selectedDocumentIndex: null,
                selectedDocument: null,
            });
        }
    };

    getTitle = () => {
        let type = '';
        if (this.props.streamType === FINANCIAL_STREAM_INCOME) { type = 'income'}
        if (this.props.streamType === FINANCIAL_STREAM_ASSET) { type = 'asset'}
        return (
            <P margin="43px 0 0 0">{`Proof of ${type}:`}</P>
        )
    };

    handleChange = event => {
        const documentRequired = this.documentsRequired;
        const index = parseInt(event.target.value);
        this.setState({
            selectedDocumentIndex: index,
            selectedDocument: documentRequired.proof_documents[index]
        });
    };

    get documentsRequired () {
        const config = this.props.config.financial_documents_validations;
        return config.find(doc => doc.income_or_asset_type === this.props.incomeOrAssetType);
    };

    getProofsLabel = () => {
        const documentRequired = this.documentsRequired;
        let proofDocuments = documentRequired.proof_documents;

        return proofDocuments.map(d => d.label).join(' + ')
    };

    render () {
        const { selectedDocumentIndex, selectedDocument } = this.state;
        const documentRequired = this.documentsRequired;
        const requireAll = documentRequired?.require_all ?? true;

        if (!documentRequired || documentRequired.proof_documents.length === 0) return null;

        return (
            <>
                {this.getTitle()}
                {requireAll || documentRequired.proof_documents.length <=1 ? (
                    <>
                        <P margin="15px 0 48px 0">{this.getProofsLabel()}</P>
                        {documentRequired.proof_documents.map((doc, index) => (
                            <ActionButton
                                key={index}
                                marginBottom={(index === documentRequired.proof_documents.length - 1) ? 68 : 17}
                                disabled={false}
                                variant="outlined"
                            >
                                Upload {doc.label}
                            </ActionButton>)
                        )}
                    </>
                ) : (
                    <>
                        <FormControl component="fieldset">
                            <RadioGroup
                                aria-label="documents"
                                name="documents"
                                value={selectedDocumentIndex}
                                onChange={this.handleChange}
                            >
                                {documentRequired.proof_documents.map((doc, index) => (
                                    <FormControlLabel
                                        key={index}
                                        value={index}
                                        control={<Radio />}
                                        label={doc.label}
                                    />
                                ))}
                            </RadioGroup>
                        </FormControl>
                        {selectedDocument && (
                            <ActionButton disabled={false} marginTop={48} marginBottom={68} variant="outlined">
                                Upload {selectedDocument.label}
                            </ActionButton>
                        )}
                    </>
                )}
            </>
        )
    }
}

UploadDocuments.propTypes = {
    incomeOrAssetType: PropTypes.number.isRequired,
    config: PropTypes.object.isRequired,
    streamType: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
    config: state.configuration,
});

export default connect(mapStateToProps)(withRelativeRoutes(UploadDocuments, ROUTES.MANUAL_INCOME_ENTRY_ADD_INCOME));
