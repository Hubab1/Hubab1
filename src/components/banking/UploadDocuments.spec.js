import { UploadDocuments } from './UploadDocuments';
import {shallow, mount} from "enzyme";
import React from "react";
import { FINANCIAL_STREAM_INCOME, FINANCIAL_STREAM_ASSET } from 'app/constants';


let defaultProps;
let store;
beforeEach(() => {
    defaultProps = {
        incomeOrAssetType: 105,
        streamType: FINANCIAL_STREAM_INCOME,
        config: {
            financial_documents_validations:
                [
                    {
                        stream_type: 5,
                        proof_documents:
                            [
                                {
                                    min_required: 1,
                                    id: 1,
                                    description: 'W2 tax form',
                                    max_required: 3,
                                    label: 'W2',
                                }
                            ],
                        lease_settings: 2,
                        require_all: false,
                        income_or_asset_type: 105,
                    },
                    {
                        stream_type: 5,
                        proof_documents:
                            [
                                {
                                    min_required: 1,
                                    id: 2,
                                    description: '1099 IRS Tax Form',
                                    max_required: 3,
                                    label: '1099 Form'
                                },
                                {
                                    min_required: 3,
                                    id: 3,
                                    description: '3 latest pay-stubs',
                                    max_required: 3,
                                    label: '3 Paystubs'
                                }
                            ],
                        lease_settings: 2,
                        require_all: false,
                        income_or_asset_type: 110,
                    },
                    {
                        stream_type: 5,
                        proof_documents:
                            [
                                {
                                    min_required: 1,
                                    id: 1,
                                    description: 'W2 tax form',
                                    max_required: 3,
                                    label: 'W2'
                                },
                                {
                                    min_required: 1,
                                    id: 2,
                                    description: '1099 IRS Tax Form',
                                    max_required: 1,
                                    label: '1099 Form'
                                }
                            ],
                        lease_settings: 2,
                        require_all: true,
                        income_or_asset_type: 115
                    }
                ],
        },
    }
});

it('Case only one required document', () => {
    const wrapper = shallow(<UploadDocuments {...defaultProps} incomeOrAssetType={105} store={store}/>);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Case multiple required document (require all)', () => {
    const wrapper = shallow(<UploadDocuments {...defaultProps} incomeOrAssetType={115} store={store}/>);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Case require one of multiple document', () => {
    const wrapper = shallow(<UploadDocuments {...defaultProps} incomeOrAssetType={110} store={store}/>);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Test title when uploading for income', () => {
    const wrapper = shallow(<UploadDocuments {...defaultProps} streamType={FINANCIAL_STREAM_INCOME} incomeOrAssetType={105} store={store} />);
    expect(wrapper.childAt(0).text()).toMatch('Proof of income:')
});

it('Test title when uploading for asset', () => {
    const wrapper = shallow(<UploadDocuments {...defaultProps} streamType={FINANCIAL_STREAM_ASSET} incomeOrAssetType={105} store={store} />);
    expect(wrapper.childAt(0).text()).toMatch('Proof of asset:')
});

it('Case no uploaded documents', () => {
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{}}
        />);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Case documents uploaded', () => {
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                // Case 1 doc
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                    ]
                },
                // Case stacked
                '2': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'def.pdf'},
                        {name: 'ghi.jpg'},
                    ]
                }
            }}
        />);
    expect(wrapper.getElement()).toMatchSnapshot();
});

it('Case minimum uploaded not met', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 3,
                                label: 'W2',
                            },
                        ],
                    lease_settings: 2,
                    require_all: false,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                    ]
                },
            }}
        />);
    expect(wrapper.find('input').length).toBe(1);
});

it('Case minimum uploaded met', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 3,
                                label: 'W2',
                            },
                        ],
                    lease_settings: 2,
                    require_all: false,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                        {name: 'def.png'},
                    ]
                },
            }}
        />);
    expect(wrapper.find('input').length).toBe(1);
});

it('Case maximum uploaded met', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 3,
                                label: 'W2',
                            },
                        ],
                    lease_settings: 2,
                    require_all: false,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                        {name: 'def.png'},
                        {name: 'ghi.png'},
                    ]
                },
            }}
        />);
    expect(wrapper.find('input').length).toBe(0);
});

it('Case require_all false and no document uploaded', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 3,
                                label: 'W2',
                            },
                            {
                                min_required: 2,
                                id: 2,
                                description: '1099 tax form',
                                max_required: 3,
                                label: '1099',
                            },
                        ],
                    lease_settings: 2,
                    require_all: false,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{}}
        />);
    expect(wrapper.find('#radioButton1').prop('disabled')).toBe(false);
    expect(wrapper.find('#radioButton2').prop('disabled')).toBe(false);
});

it('Case require_all false and one of documents types uploaded, disable the rest', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 3,
                                label: 'W2',
                            },
                            {
                                min_required: 2,
                                id: 2,
                                description: '1099 tax form',
                                max_required: 3,
                                label: '1099',
                            },
                        ],
                    lease_settings: 2,
                    require_all: false,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                    ]
                },
            }}
        />);
    expect(wrapper.find('#radioButton1').prop('disabled')).toBe(false);
    expect(wrapper.find('#radioButton2').prop('disabled')).toBe(true);
});

it('Case require_all is true and only one document type met minimum', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 2,
                                label: 'W2',
                            },
                            {
                                min_required: 1,
                                id: 2,
                                description: '1099 tax form',
                                max_required: 3,
                                label: '1099',
                            },
                        ],
                    lease_settings: 2,
                    require_all: true,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                        {name: 'def.jpg'},
                    ]
                },
            }}
        />);
    expect(wrapper.find('input').length).toBe(1);
});


it('Case require_all is true and all documents types met min and max requirement', () => {
    const config = {
        financial_documents_validations:
            [
                {
                    stream_type: 5,
                    proof_documents:
                        [
                            {
                                min_required: 2,
                                id: 1,
                                description: 'W2 tax form',
                                max_required: 2,
                                label: 'W2',
                            },
                            {
                                min_required: 1,
                                id: 2,
                                description: '1099 tax form',
                                max_required: 1,
                                label: '1099',
                            },
                        ],
                    lease_settings: 2,
                    require_all: true,
                    income_or_asset_type: 105,
                },
            ],
    };
    const wrapper = shallow(
        <UploadDocuments
            {...defaultProps}
            config={config}
            streamType={FINANCIAL_STREAM_ASSET}
            incomeOrAssetType={105}
            store={store}
            uploadedDocuments={{
                '1': {
                    id:1,
                    label: 'W2',
                    files:[
                        {name: 'abc.png'},
                        {name: 'def.jpg'},
                    ]
                },
                '2': {
                    id: 2,
                    label: '1099',
                    files:[
                        {name: 'abc.png'},
                    ]
                },
            }}
        />);
    expect(wrapper.find('input').length).toBe(0);
});
