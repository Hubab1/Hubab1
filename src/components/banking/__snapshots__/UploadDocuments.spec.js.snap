// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case documents uploaded 1`] = `
<React.Fragment>
  <ForwardRef(render)
    margin="43px 0 0 0"
  >
    Proof of asset:
  </ForwardRef(render)>
  <React.Fragment>
    <ForwardRef(render)
      margin="15px 0 48px 0"
    >
      W2
    </ForwardRef(render)>
    <ForwardRef(render)>
      <div
        className="uploaded-document"
      >
        <div
          className="uploaded-document-title"
        >
          W2
        </div>
        <div
          className="uploaded-document-filename"
        >
          <ForwardRef(render)>
            abc.png
          </ForwardRef(render)>
        </div>
      </div>
      <div
        className="uploaded-document"
      >
        <div
          className="uploaded-document-title"
        >
          W2
        </div>
        <div
          className="uploaded-document-filename"
        >
          <ForwardRef(render)>
            def.pdf
          </ForwardRef(render)>
        </div>
        <div
          className="uploaded-document-filename"
        >
          <ForwardRef(render)>
            ghi.jpg
          </ForwardRef(render)>
        </div>
      </div>
    </ForwardRef(render)>
    <ForwardRef(render)
      marginBottom={51}
      marginTop={48}
    >
      <ForwardRef(WithStyles)
        classes={
          Object {
            "label": "css-1405lf9",
            "root": "css-11ldqzb",
          }
        }
        color="primary"
        component="label"
        fullWidth={true}
        variant="outlined"
      >
        Upload 
        W2
        <input
          accept="image/*,.pdf"
          id="1"
          name="1"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </ForwardRef(WithStyles)>
    </ForwardRef(render)>
  </React.Fragment>
</React.Fragment>
`;

exports[`Case multiple required document (require all) 1`] = `
<React.Fragment>
  <ForwardRef(render)
    margin="43px 0 0 0"
  >
    Proof of income:
  </ForwardRef(render)>
  <React.Fragment>
    <ForwardRef(render)
      margin="15px 0 48px 0"
    >
      W2 + 1099 Form
    </ForwardRef(render)>
    <ForwardRef(render)
      marginBottom={51}
      marginTop={48}
    >
      <ForwardRef(WithStyles)
        classes={
          Object {
            "label": "css-1405lf9",
            "root": "css-11ldqzb",
          }
        }
        color="primary"
        component="label"
        fullWidth={true}
        variant="outlined"
      >
        Upload 
        W2
        <input
          accept="image/*,.pdf"
          id="1"
          name="1"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        classes={
          Object {
            "label": "css-1405lf9",
            "root": "css-11ldqzb",
          }
        }
        color="primary"
        component="label"
        fullWidth={true}
        variant="outlined"
      >
        Upload 
        1099 Form
        <input
          accept="image/*,.pdf"
          id="2"
          name="2"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </ForwardRef(WithStyles)>
    </ForwardRef(render)>
  </React.Fragment>
</React.Fragment>
`;

exports[`Case no uploaded documents 1`] = `
<React.Fragment>
  <ForwardRef(render)
    margin="43px 0 0 0"
  >
    Proof of asset:
  </ForwardRef(render)>
  <React.Fragment>
    <ForwardRef(render)
      margin="15px 0 48px 0"
    >
      W2
    </ForwardRef(render)>
    <ForwardRef(render) />
    <ForwardRef(render)
      marginBottom={51}
      marginTop={48}
    >
      <ForwardRef(WithStyles)
        classes={
          Object {
            "label": "css-1405lf9",
            "root": "css-11ldqzb",
          }
        }
        color="primary"
        component="label"
        fullWidth={true}
        variant="outlined"
      >
        Upload 
        W2
        <input
          accept="image/*,.pdf"
          id="1"
          name="1"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </ForwardRef(WithStyles)>
    </ForwardRef(render)>
  </React.Fragment>
</React.Fragment>
`;

exports[`Case only one required document 1`] = `
<React.Fragment>
  <ForwardRef(render)
    margin="43px 0 0 0"
  >
    Proof of income:
  </ForwardRef(render)>
  <React.Fragment>
    <ForwardRef(render)
      margin="15px 0 48px 0"
    >
      W2
    </ForwardRef(render)>
    <ForwardRef(render)
      marginBottom={51}
      marginTop={48}
    >
      <ForwardRef(WithStyles)
        classes={
          Object {
            "label": "css-1405lf9",
            "root": "css-11ldqzb",
          }
        }
        color="primary"
        component="label"
        fullWidth={true}
        variant="outlined"
      >
        Upload 
        W2
        <input
          accept="image/*,.pdf"
          id="1"
          name="1"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
      </ForwardRef(WithStyles)>
    </ForwardRef(render)>
  </React.Fragment>
</React.Fragment>
`;

exports[`Case require one of multiple document 1`] = `
<React.Fragment>
  <ForwardRef(render)
    margin="43px 0 0 0"
  >
    Proof of income:
  </ForwardRef(render)>
  <React.Fragment>
    <ForwardRef(WithStyles)
      component="fieldset"
    >
      <ForwardRef(RadioGroup)
        aria-label="documents"
        name="documents"
        onChange={[Function]}
        value={null}
      >
        <ForwardRef(WithStyles)
          control={<ForwardRef(WithStyles) />}
          label="1099 Form"
          value={0}
        />
        <ForwardRef(WithStyles)
          control={<ForwardRef(WithStyles) />}
          label="3 Paystubs"
          value={1}
        />
      </ForwardRef(RadioGroup)>
    </ForwardRef(WithStyles)>
  </React.Fragment>
</React.Fragment>
`;
